### PR TITLE
concentrated - revert() functions to reduce size 

### PR DESCRIPTION
**Pull request by z0r0z**
~~

### CODE REVIEW

File: contracts/libraries/concentratedPool/FullMath.sol [hunk 1]
```diff
@@ -32,15 +37,15 @@ library FullMath {
             }
             // @dev Handle non-overflow cases, 256 by 256 division.
             if (prod1 == 0) {
-                require(denominator > 0);
+                if (denominator == 0) revert ZeroDenom();
```
Comment (maxsam4) [inline contracts/libraries/concentratedPool/FullMath.sol:40]: This just increases gas cost. Undo

File: contracts/libraries/concentratedPool/FullMath.sol [hunk 2]
```diff
@@ -32,15 +37,15 @@ library FullMath {
...
                 assembly {
                     result := div(prod0, denominator)
                 }
                 return result;
             }
             // @dev Make sure the result is less than 2**256 -
             // also prevents denominator == 0.
-            require(denominator > prod1);
+            if (denominator < prod1) revert InsuffDenom();
```
Comment (maxsam4) [inline contracts/libraries/concentratedPool/FullMath.sol:48]: undo

File: contracts/libraries/concentratedPool/FullMath.sol [hunk 3]
```diff
@@ -113,7 +118,7 @@ library FullMath {
         result = mulDiv(a, b, denominator);
         unchecked {
             if (mulmod(a, b, denominator) != 0) {
-                require(result < type(uint256).max);
+                if (result > type(uint256).max) revert ResultMax();
```
Comment (maxsam4) [inline contracts/libraries/concentratedPool/FullMath.sol:121]: This doesn't work. you can never go above max. it'll wrap around.

Can we continue using requires?

Comment (z0r0z) [reply contracts/libraries/concentratedPool/FullMath.sol:121]: that would be my preference @maxsam4 maybe we can treat this PR as something that could save space if we run into limits

**Pull request closed without merging**